engine/gamesys/src/gamesys/scripts/script_buffer.cpp

```diff
@@ -879,6 +880,209 @@ namespace dmGameSystem
+            if (!lua_isnumber(L,4)) {
+                return DM_LUA_ERROR("%s.%s invalid value type supplied: %ld.", SCRIPT_LIB_NAME, SCRIPT_TYPE_NAME_BUFFERMETADATA, valueTypeRaw);
+            }
+            dmBuffer::ValueType valueType = (dmBuffer::ValueType) valueTypeRaw;
+
+            // get array of values
+            luaL_checktype(L, 3, LUA_TTABLE);
+            uint32_t count = lua_objlen(L, 3);
+            if (count > 0) {
+                void* values;
+                lua_pushvalue(L, 3);
+
+                switch(valueType) {
+                    case dmBuffer::VALUE_TYPE_FLOAT32:
+                        values = (void*) luaTableToArray<float>(L, count, valueType);
+                    break;
+                    case dmBuffer::VALUE_TYPE_UINT8:
+                        values = (void*) luaTableToArray<uint8_t>(L, count, valueType);
+                    break;
+                    case dmBuffer::VALUE_TYPE_UINT16:
+                        values = (void*) luaTableToArray<uint16_t>(L, count, valueType);
+                    break;
+                    case dmBuffer::VALUE_TYPE_UINT32:
+                        values = (void*) luaTableToArray<uint32_t>(L, count, valueType);
+                    break;
+                    case dmBuffer::VALUE_TYPE_UINT64:
+                        values = (void*) luaTableToArray<uint64_t>(L, count, valueType);
+                    break;
+                    case dmBuffer::VALUE_TYPE_INT8:
+                        values = (void*) luaTableToArray<int8_t>(L, count, valueType);
+                    break;
+                    case dmBuffer::VALUE_TYPE_INT16:
+                        values = (void*) luaTableToArray<int16_t>(L, count, valueType);
+                    break;
+                    case dmBuffer::VALUE_TYPE_INT32:
+                        values = (void*) luaTableToArray<int32_t>(L, count, valueType);
+                    break;
+                    case dmBuffer::VALUE_TYPE_INT64:
+                        values = (void*) luaTableToArray<int64_t>(L, count, valueType);
+                    break;
+                    default:
+                        return DM_LUA_ERROR("%s.%s invalid value type supplied: %ld.", SCRIPT_LIB_NAME, SCRIPT_TYPE_NAME_BUFFERMETADATA, valueTypeRaw);
+                }
+
+                dmBuffer::Result r = dmBuffer::SetMetaData(hbuffer, entry_name, (void*) values, count, valueType);
+                free(values);
+
+                if( r != dmBuffer::RESULT_OK )
+                {
+                    return DM_LUA_ERROR("%s.%s: cannot set metadata for buffer: %s", SCRIPT_LIB_NAME, SCRIPT_TYPE_NAME_BUFFERMETADATA, dmBuffer::GetResultString(r));
+                }
+            } else {
+                return DM_LUA_ERROR("%s.%s: invalid metadata", SCRIPT_LIB_NAME, SCRIPT_TYPE_NAME_BUFFERMETADATA);
+            }
+        } else
+            return DM_LUA_ERROR("%s.%s: invalid number of arguments", SCRIPT_LIB_NAME, SCRIPT_TYPE_NAME_BUFFERMETADATA);
```

Comment:
I prefer if we use { and } even on a one-liner. Like you did on line 984-986

Reply:
Also, I'm not sure we actually need to check that there are 4 or more values on the stack? Let's say there's only three then `luaL_checkinteger(L, 4);` will result in a Lua error with a message saying that it expected a value of type int as the fourth argument.

Or is this because we need to use DM_LUA_ERROR since we have DM_LUA_STACK_CHECK above? I don't think that it is needed if a luaL_checkfoo generates a Lua error. Correct or not @JCash?

Reply:
Correct. I've removed the argument count check. I'll wait for Mathias to confirm before commiting.

Reply:
I'll commit but leave this conversation open in case we need to revisit later.